plugins/jira-dashboard-backend/src/filters.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Config } from '@backstage/config';
```

Comment:
This file if it is to stay as it is should move to the frontend.
However these default filters will be very specific to each orgiznation using it, and so i think we should try make it more configurable.
It would also be nice to have different filter for different types.
Maybe we could set them in app-config or pass as a parameter in EntityPage
`      if={isJiraDashboardAvailable}
      path="/jira-dashboard"
      title="Jira Dashboard"
    >
      <EntityJiraTable filter="12345" />
      <EntityJiraTable query="status = New" orderBy="DESC" filterByProject:true filterByComponent:false />
      <EntityJiraDashboardContent /> // Show annotation filters just`

Let me know what you think about this, maybe it's something we can work on together

Reply:
Got what i suggested above working locally with the following updates.
if defaultFilters are not included in app-config then we use the existing filters
app-config
```json
jira:
  token: "Basic ${JIRA_TOKEN}"
  baseUrl: "${JIRA_URL}/rest/api/2/"
  # userEmailSuffix: "@abc.com"
  defaultFilters:
    - name: 'Ericsson Open Issues'
      shortName: 'OPEN'
      query: 'resolution = Unresolved'
    - name: 'Ericsson Assigned to me'
      shortName: 'ME'
      query: 'resolution = Unresolved'
      filterOnUser: true
      order: "ASC"
````
filters.ts
```
  const jiraUser = `${username}${resolveUserEmailSuffix(config)}`
  const defaultFiltersConfig: Config[] | undefined = resolveJiraFilters(config)
  
  if (defaultFiltersConfig) {
    const filterArray: Filter[] = []
    for (const filter of defaultFiltersConfig) {
      const filterOnUser = (filter.getOptionalBoolean('filterOnUser') ? `assignee = ${jiraUser} AND ` : "")
      const order: string = (`ORDER BY updated ${filter.getOptionalString('order') || "DESC"}`)
      const jiraFilter: Filter = {
        name: filter.getString('name'),
        query: `${filterOnUser} ${filter.getString('query')} ${order}`,
        shortName: filter.getString('shortName')
      }
      filterArray.push(jiraFilter)
    }
    return filterArray
  }
  const assignedToMeFilter: Filter = {
    name: 'Assigned to me',
    shortName: 'ME',
    query: `assignee = "${jiraUser}" AND resolution = Unresolved ORDER BY updated DESC`,
  };
  return [openFilter, incomingFilter, assignedToMeFilter];
```

config.ts
```
export function resolveJiraFilters(config: Config): Config[] | undefined {
  try {
    return config.getOptionalConfigArray(JIRA_FILTERS);
    } catch (error) {
    throw new Error(`Invalid Jira defaultFilters, ${error}`);
  }
}
```
Let me know what you think of above, Maybe this can be a seperate PR later

Reply:
FYI updated code above to include a filterOnComponent option